File: src/utils/makeupSuggestions.ts
import { SkinProfile, MakeupSuggestion } from '../types';
import { foundations, concealers, colorCorrectors } from '../data/products';

export function getSuggestions(profile: SkinProfile): MakeupSuggestion[] {
  const suggestions: MakeupSuggestion[] = [];

  // Foundation suggestions
  const suitableFoundations = foundations.filter(foundation => {
    return foundation.suitableFor.skinTypes.includes(profile.skinType) &&
           (!foundation.suitableFor.sensitivity || 
            foundation.suitableFor.sensitivity.includes(profile.sensitivity));
  });

  if (suitableFoundations.length > 0) {
    const bestMatch = suitableFoundations[0];
    suggestions.push({
      category: 'Foundation',
      productType: 'Foundation',
      recommendation: `${bestMatch.brand} ${bestMatch.name} (${bestMatch.price}) - ${bestMatch.description}`,
      reason: `Perfect for ${profile.skinType} skin with ${profile.sensitivity} sensitivity. ${bestMatch.link}`,
    });
  }

  // Concealer suggestions
  const suitableConcealers = concealers.filter(concealer => {
    return concealer.suitableFor.skinTypes.includes(profile.skinType) &&
           (profile.concerns.some(concern => 
             concealer.suitableFor.concerns?.includes(concern)
           ));
  });

  if (suitableConcealers.length > 0) {
    const bestMatch = suitableConcealers[0];
    suggestions.push({
      category: 'Concealer',
      productType: 'Concealer',
      recommendation: `${bestMatch.brand} ${bestMatch.name} (${bestMatch.price}) - ${bestMatch.description}`,
      reason: `Ideal for your skin type and concerns. ${bestMatch.link}`,
    });
  }

  // Color corrector suggestions based on concerns
  if (profile.concerns.includes('Redness')) {
    const colorCorrector = colorCorrectors.find(cc => 
      cc.suitableFor.concerns?.includes('Redness')
    );
    if (colorCorrector) {
      suggestions.push({
        category: 'Color Corrector',
        productType: 'Color Corrector',
        recommendation: `${colorCorrector.brand} ${colorCorrector.name} (${colorCorrector.price}) - ${colorCorrector.description}`,
        reason: `Helps neutralize redness before foundation application. ${colorCorrector.link}`,
      });
    }
  }

  return suggestions;
}

File: utils/makeup_suggestions.py
from models import SkinProfile, MakeupSuggestion
from data.products import foundations, concealers, color_correctors

def get_suggestions(profile: SkinProfile) -> list[MakeupSuggestion]:
    suggestions = []

    # Foundation suggestions
    suitable_foundations = [
        f for f in foundations
        if profile.skin_type in f['suitable_for']['skin_types']
        and (not f['suitable_for'].get('sensitivity')
             or profile.sensitivity in f['suitable_for']['sensitivity'])
    ]

    if suitable_foundations:
        best_match = suitable_foundations[0]
        suggestions.append(MakeupSuggestion(
            category='Foundation',
            product_type='Foundation',
            recommendation=f"{best_match['brand']} {best_match['name']} ({best_match['price']}) - {best_match['description']}",
            reason=f"Perfect for {profile.skin_type} skin with {profile.sensitivity} sensitivity. {best_match['link']}"
        ))

    # Concealer suggestions
    suitable_concealers = [
        c for c in concealers
        if profile.skin_type in c['suitable_for']['skin_types']
        and any(concern in c['suitable_for'].get('concerns', [])
               for concern in profile.concerns)
    ]

    if suitable_concealers:
        best_match = suitable_concealers[0]
        suggestions.append(MakeupSuggestion(
            category='Concealer',
            product_type='Concealer',
            recommendation=f"{best_match['brand']} {best_match['name']} ({best_match['price']}) - {best_match['description']}",
            reason=f"Ideal for your skin type and concerns. {best_match['link']}"
        ))

    # Color corrector suggestions
    if 'Redness' in profile.concerns:
        color_corrector = next(
            (cc for cc in color_correctors
             if 'Redness' in cc['suitable_for'].get('concerns', [])),
            None
        )
        if color_corrector:
            suggestions.append(MakeupSuggestion(
                category='Color Corrector',
                product_type='Color Corrector',
                recommendation=f"{color_corrector['brand']} {color_corrector['name']} ({color_corrector['price']}) - {color_corrector['description']}",
                reason=f"Helps neutralize redness before foundation application. {color_corrector['link']}"
            ))

    return suggestions